File: apps/ios/BlessedIrembo/BlessedIrembo/Utilities/Language.swift
/// Supported Languages
///
/// Enum defining the supported languages in the Blessed Irembo application.
/// Provides helper properties for display names and system integration.

import Foundation

enum Language: String, CaseIterable, Identifiable {
    case english = "en"
    case kinyarwanda = "rw"
    
    var id: String { self.rawValue }
    
    var code: String { self.rawValue }
    
    var displayName: String {
        switch self {
        case .english: return "EN"
        case .kinyarwanda: return "RW"
        }
    }
    
    var fullName: String {
        switch self {
        case .english: return "English"
        case .kinyarwanda: return "Kinyarwanda"
        }
    }
}

File: apps/android/BlessedIrembo/app/src/main/java/com/blessedirembo/app/util/Language.kt
package com.blessedirembo.app.util

/**
 * Language
 *
 * Enum defining the supported languages in the Blessed Irembo application.
 * Mirrors iOS Language.swift exactly.
 */
enum class Language(val code: String) {
    ENGLISH("en"),
    KINYARWANDA("rw");

    /** Short display label shown in the FloatingLanguageSwitcher pill (e.g. "EN") */
    val displayName: String
        get() = when (this) {
            ENGLISH -> "EN"
            KINYARWANDA -> "RW"
        }

    /** Full language name (e.g. "English") */
    val fullName: String
        get() = when (this) {
            ENGLISH -> "English"
            KINYARWANDA -> "Kinyarwanda"
        }

    companion object {
        /** Parse a language code string ("en", "rw") back to a Language, defaulting to ENGLISH. */
        fun fromCode(code: String): Language =
            entries.firstOrNull { it.code == code } ?: ENGLISH
    }
}
